### DISCUSSION TITLE
How to extract text and values from a table with no borders?

### DISCUSSION THREAD

**Discussion by bballamudi** [jsvine/pdfplumber#633] [Ask for help with specific PDFs]
Hello all,

Many regards for developing pdfplumber it is very useful. I need to extract text from a table with no borders in a pdf file. I was able to get it with other tables with borders by setting the parameters around the border specifics. Can someone help me with a peculiar case of a text that mimics a table but has got no borders or lines as show below.

![image](https://user-images.githubusercontent.com/21183667/160512134-cead1022-8b03-47cc-b2cf-1953740a9fa2.png)

**Comment by jsvine** [comment]
Hi @bballamudi, and I'm glad to hear that you've found the library to be useful. Without the actual PDF being attached, it is somewhat difficult to make suggestions, but I'd recommend thinking of what you see in the screenshot *not as a table*, but rather more generally as *structured information*. What rules can you ascertain re. how the PDF layout was created? 

For example, to extract the total cost+fee number (`$4,600,000,000.00 (EST.)` in this screenshot), you might look for the black horizontal line, and then select the text directly below it.

**Reply by bballamudi** [reply]
@jsvine Thanks for your reply. How do I look fro horizontal line and then select the text directly below it. Can you guide me code that part up?

**Reply by bballamudi** [reply]
[DOD_RFP_Contract (1).pdf](https://github.com/jsvine/pdfplumber/files/8379957/DOD_RFP_Contract.1.pdf)

Kindly find the pdf file attached. I am trying to find the key value pairs of the highlighted entities.

**Reply by jsvine** [reply]
Can you provide a version of the PDF that has not been annotated? I might be able provide some general guidance, and possibly some limited sample code.

**Reply by bballamudi** [reply]
[DOD_RFP_Contract.pdf](https://github.com/jsvine/pdfplumber/files/8385243/DOD_RFP_Contract.pdf)

This is all I have. I don't know how to remove the highlighted parts though. Do you think this helps?

**Reply by jsvine** [reply]
This helps, thank you. Taking the example I mentioned above about the text below the line, you could do something like this: https://notebooksharing.space/view/d69ded3596437271272b38bd9f3023605ec8647216a39be2692a4dfe3bb729b3

Also attaching the notebook here in case URL stops working in the future: [discussion-633.ipynb.txt](https://github.com/jsvine/pdfplumber/files/8389721/discussion-633.ipynb.txt)
